### PR TITLE
Add a missing check for a corner case with package use site / internal decl site

### PR DESCRIPTION
**Pull request by elsh**
* Add a missing check for a corner case with package use site / internal decl site
* Add package keyword to the code completion override lookup
* Update tests
Resolves rdar://106732804, rdar://104987455

### CODE REVIEW

**Comment by elsh** [comment]
@swift-ci smoke test

**Comment by elsh** [comment]
@swift-ci smoke test

File: test/Sema/accessibility.swift [hunk 1]
```diff
@@ -1471,7 +1471,11 @@ public class DerivedFromInternalConcreteGenericComposition : InternalConcreteGen
   public func publicReq() {}
 }
 
-// FIXME: rdar://104987455 should have expected note and error 'class cannot be declared public because its superclass is internal'
```
Comment (elsh) [inline test/Sema/accessibility.swift:1474]: turns out this was expected; added a test using `fileprivate` typealias from `internal` decls to illustrate this

File: test/Sema/accessibility_package.swift [hunk 2]
```diff
@@ -1,34 +1,18 @@
 // RUN: %empty-directory(%t)
-// RUN: %{python} %utils/split_file.py -o %t %s
+// RUN: split-file %s %t
```
Comment (elsh) [inline test/Sema/accessibility_package.swift:2]: No functional changes in this file; just a style change

File: test/Sema/accessibility_package_inline.swift [hunk 3]
```diff
@@ -1,53 +1,22 @@
 // RUN: %empty-directory(%t)
-// RUN: %{python} %utils/split_file.py -o %t %s
+// RUN: split-file %s %t
```
Comment (elsh) [inline test/Sema/accessibility_package_inline.swift:2]: No functional changes in this file; just a style change

File: test/attr/accessibility_proto.swift [hunk 4]
```diff
@@ -243,3 +243,29 @@ package protocol PkgEmptyInit {
 public struct PkgBuggy: PkgEmptyInit {
   // expected-error@-1 {{initializer 'init()' must be declared package because it matches a requirement in package protocol 'PkgEmptyInit'}}
 }
+
+package protocol PkgReqProvider {}
```
Comment (elsh) [inline test/attr/accessibility_proto.swift:247]: Tests the source file change in this PR

**Comment by elsh** [comment]
@swift-ci test

**Comment by elsh** [comment]
@swift-ci smoke test

**Review by xymus — APPROVED** [review]
Thank you for the test style updates!

**Comment by xymus** [comment]
The macOS test failed on something likely unrelated:
```
3.	Running pass 'Function Pass Manager' on module '/Users/ec2-user/jenkins/workspace/swift-PR-macos/branch-main/build/buildbot_incremental/swift-macosx-x86_64/stdlib/public/StringProcessing/WATCHOS/armv7k/_StringProcessing.o'.
4.	Running pass 'Control Flow Optimizer' on function '@"$s17_StringProcessing9ProcessorV8matchSeq_12isScalarModeSbSs_SbtF"'
```

@swift-ci please test macOS

**Comment by elsh** [comment]
@swift-ci please test macOS

**Pull request merged**